README.md

```diff
@@ -1,16 +1,413 @@
+/*
+Happy Birthday, Odin One-Eye!
+From Asgard to Nifelheim, you're the best all-father ever.
+
+Love,
+Admiration, respect, and love, Thor
+Your son, Loki
+*/
+```
+
+Many JavaScript developers even go so far to call the variable we called
+`outerContext` by the name `self` which sure is confusing for Ruby programmers!
+In any case, by using an assignment with `let`, `var` or `const` we put the
+original context within the function-level scope that the inner function
+encloses (as a closure). This means inside the inner function, we can get
+"back" to the outer function's context. That's solution number two.
+
+What we would _really_ like is for there to be a way to tell the `function`
+inside of `forEach` to
+
+1. _Not_ declare its own context **but also**
+2. _Not_ require us to do some extra work with using `bind` or a `thisArg`.
+
+In ES6, JavaScript gave us an answer: the "arrow function expression." This is
+our third, and, as of 2019, the most popular option. Nevertheless, you will see
+all the other approaches used in framework code (e.g. React) and in other
+codebases.
+
+## Solution 3: Uses an Arrow Function Expression to Create a Function Without Its Own Context
+
+The arrow function expression is yet another way of writing a function
+expression. They look different from "old style" function expressions, but the
+***most important difference*** is that the arrow function ***does inherit the
+context of its parent containing `function`***. An arrow function looks like
+this:
+
+```js
+// The let greeter is merely the assignment, the expression begins at `(`
+let greeter = (nameToGreet) => {
+    let messasge = `Good morning ${nameToGreet}`
+    console.log(message)
+    return "Greeted: " + nameToGreet
+}
+let result = greeter("Max") //=> "Greeted: Max"
+```
+
+Which, excluding context-switching differences, is the exact same as:
+
+```js
+let greeter = function(nameToGreet) {
+    let messasge = `Good morning ${nameToGreet}`
+    console.log(message)
+    return "Greeted: " + nameToGreet
+}
+let result = greeter("Max Again") //=> "Greeted: Max Again"
+```
+
+Because arrow functions are _so often used_ to take a value, do a single
+operation with it, and return the result, arrow functions have two shortcuts:
+
+* If you pass only one argument, you don't have to wrap the single parameter in `()`
+* If there is only one expression, you don't need to wrap it in `{}` and the result of that expression is automatically returned.
+* Anti-Shortcut: If you *DO* use `{}`, you must explicitly `return` the return value
+
+As a result, this longer expression (with context shift):
+
+```js
+[1,2,3].reduce(function(memo, elem){
+    return memo + elem * 10
+}, 0) //=> 60
+```
+
+Can become this much shorter call with ***no*** context shift:
+
+```js
+[1,2,3].reduce((memo, elem) => memo + elem * 10, 0) //=> 60
+```
+
+Thus Thor and Loki can fix their problem and wish their father a happy
+birthday most elegantly with the following code:
```

Comment:
I _think_ we want to see this example of fixing the context bug before we get in to the other details around arrow functions - StudentStackLevelTooDeep